src/mesh/exodusII_io_helper.C

```diff
@@ -1175,6 +1175,10 @@ void ExodusII_IO_Helper::read_elem_in_block(int block)
               const int * as_int = static_cast<int *>(attr.values);
               std::copy(as_int, as_int+attr.value_count, bex_elem_degrees.begin());
 
+              // ex_get_attributes did a values=calloc(); free() is our job.
+              if (attr.values)
+                free(attr.values);
```

Comment:
This is... real ugly. Is this even documented anywhere? I didn't see anything in v8.11/exodus/include/exodusII.h

Reply:
It is **real** ugly.  It's documented on ex_get_attribute.c in a *comment* (not a Doxygen comment even) on line 270.

Exodus is under active development right now, with a friendly + responsive maintainer, so we should definitely see if things are clarified in the latest hash, and if not then file an issue (or just a PR fixing the documentation, maybe with a comment on the PR suggesting an ex_free_attribute(s) API for forward compatibility?).

I didn't want to fire something off before I could be diplomatic about it, though; my initial sarcastic reactions on Slack or even "real ugly" might not have been conducive toward "friendly + responsive" remaining true in the long term.  ;-)  Even if it wasn't taken personally, there's been a bunch of good work in ExodusII recently and there are some new features on the horizon we could really use later, and I don't mind some short-term underdocumentation if that's the price of rapid development.  I've certainly been more than guilty of making that tradeoff myself...

Reply:
Yeah I agree that your fix is the needed one, and that `ex_free_attribute()` would be preferable, though still not ideal. If you are creating issues, then this "not accessed" comment in the header is also misleading:
```
typedef struct ex_attribute
{
  ex_entity_type entity_type;
  int64_t        entity_id;
  char           name[NC_MAX_NAME];
  ex_type        type; /* int, double, text */
  size_t         value_count;
  void *         values; /* not accessed if NULL */
} ex_attribute;
```

Maybe wrapping `ex_attribute` in some kind of RAII class could also work, though I don't have time to devote to this.